anesthetic/samples.py

```diff
@@ -152,16 +152,17 @@ def plot(self, ax, paramname_x, paramname_y=None, *args, **kwargs):
         do_1d_plot = paramname_y is None or paramname_x == paramname_y
         kwargs['label'] = kwargs.get('label', self.label)
         ncompress = kwargs.pop('ncompress', None)
+        ncompress_1d = kwargs.pop('ncompress_1d', ncompress)
```

Comment:
I'm not sure whether introducing `ncompress_1d` is the best way of implementing this feature. If you have a better idea, please let me know, @williamjameshandley.

Problem is, we need to be able to adjust the 1d kde compression independently from the 2d compression...

Reply:
OK, this is actually quite subtle, but I think we can fix this with only very small changes to the code.

The issue is that ncompress is used to downsample the kde (to speed things up) to the points `x`, but that those same points are then the ones that are used to evaluate the kde for plotting. There is no reason that these points need to be the same (and in fact for `kde_contour_plot_2d` they are not). It would be better practice anyway to separate them into `x_compress` and `x_plot`.

In 1D, one option would be to evaluate the kde on `x_plot` a uniform grid from xmin to xmax. This however may create issues for heavily compressed posteriors as 
1. you may not get resolution within the posterior bulk
2. this will mess around with the plotting limits, defaulting them to xmin and xmax

Point 1 can be fixed by using something like
 ```python
x_plot = sorted(np.concatenate([x_compress,np.linspace(xmin,xmax,ncompress]))
```
and this is in keeping with how `kde_contour_plot_2d` does it as well.

Point 2 is a little harder, and may require experimentation, but I imagine that following `kde_contour_plot_2d`'s lead again is the way to go, which is to base the bounds on `x_compress`, rather than `x_plot`

Reply:
> It would be better practice anyway to separate them into `x_compress` and `x_plot`.

This actually also applies to `xmin` and `xmax`, doesn't it? This could be related to the issue that I raised in #104 , where changing the limits in a way that the posterior bulk is cut messes with the plot.

Reply:
It does. However, provided xmin and xmax are larger than the posterior bulk, it should behave consistently with respect to the rest of the code. I think it would be nice to complete this PR first, and then for all of #104 #95 #93 to be addressed at the same time.

Reply:
The envisaged split into `x_compress` and `x_plot` does not work as hoped. It is fine in the 2d case, where we don't try to plot the tails. However, if `x_compress` does not contain the tail region, then it also won't plot anything in that region even if `x_plot` requests it.

The latest commit [bc0b9e7](https://github.com/williamjameshandley/anesthetic/pull/99/commits/bc0b9e73c43e16554b73bf4f5d59852191fda9f7) seems to work, needs a thorough check, whether that is allowed, though.